RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/PixelClustering.h

```diff
@@ -0,0 +1,451 @@
+#ifndef RecoLocalTracker_SiPixelClusterizer_alpaka_PixelClustering_h
+#define RecoLocalTracker_SiPixelClusterizer_alpaka_PixelClustering_h
+
+#include <cmath>
+#include <cstdint>
+#include <cstdio>
+#include <type_traits>
+#include <alpaka/alpaka.hpp>
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/HistoContainer.h"
+#include "DataFormats/SiPixelClusterSoA/interface/ClusteringConstants.h"
+#include "Geometry/CommonTopologies/interface/SimplePixelTopology.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/SimpleVector.h"
+
+// #define GPU_DEBUG
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  namespace pixelClustering {
+
+#ifdef GPU_DEBUG
+    template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+    ALPAKA_STATIC_ACC_MEM_GLOBAL uint32_t gMaxHit = 0;
+#endif
+
+    namespace PixelStatus {
+      // Phase-1 pixel modules
+      constexpr uint32_t pixelSizeX = pixelTopology::Phase1::numRowsInModule;
+      constexpr uint32_t pixelSizeY = pixelTopology::Phase1::numColsInModule;
+
+      // Use 0x00, 0x01, 0x03 so each can be OR'ed on top of the previous ones
+      enum Status : uint32_t { kEmpty = 0x00, kFound = 0x01, kDuplicate = 0x03 };
```

Comment:
I think using an `enum` is cleaner, because it defines a type.
Otherwise we would use `uint32_t` in place of `Status`, making it less clear what variables are supposed to hold only one of those values.